internal/loader/configuration_client_manager.go

```diff
@@ -0,0 +1,478 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package loader
+
+import (
+	"azappconfig/provider/internal/properties"
+	"context"
+	"fmt"
+	"math"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	acpv1 "azappconfig/provider/api/v1"
+
+	"math/rand"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig"
+	"github.com/google/uuid"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+//go:generate mockgen -destination=mocks/mock_configuration_client_manager.go -package mocks . ClientManager
+
+type ConfigurationClientManager struct {
+	ReplicaDiscoveryEnabled   bool
+	SpecifiedClient           *azappconfig.Client
```

Comment:
How about call it `StaticClients` and make it to be an array? To potentially support customer can pass more than one static clients.